packages/cactus-api-client/src/main/typescript/socketio-api-client.ts

```diff
@@ -26,40 +26,31 @@ import { finalize } from "rxjs/operators";
 /**
  * Default logic for validating responses from socketio connector (validator).
  * Assumes that message is JWT signed with validator private key.
- * @param keyPath - Absolute or relative path to validator public key.
+ * @param publicKey - Validator public key.
  * @param targetData - Signed JWT message to be decoded.
  * @returns Promise resolving to decoded JwtPayload.
  */
 export function verifyValidatorJwt(
-  keyPath: string,
+  publicKey: string,
```

Comment:
@outSH Does this break code that uses the 1.0.0 version with a file path instead of the public key as a string?

Reply:
@petermetz `verifyValidatorJwt` is exported just for unit-testing purposes, I don't think there's a use case where a client would be concerned with this function.

Reply:
> verifyValidatorJwt is exported just for unit-testing purposes

@outSH Can you (eventually) move the whole thing into one of the test packages then?

> I don't think there's a use case where a client would be concerned with this function

I would still recommend practicing good sem-ver hygiene just to get into the habit. My assumption is that there's a tiny effort associated to this. If that assumption is incorrect, then it's a different story of course. Let me know.

Reply:
> Can you (eventually) move the whole thing into one of the test packages then?

@petermetz It's not a test-util. It's exported separately to simplify unit test of message verification functionality. It doesn't make sense outside of `SocketIOApiClient`, but testing it through an ApiClient would greatly convolute the test.

> I would still recommend practicing good sem-ver hygiene just to get into the habit. My assumption is that there's a tiny effort associated to this. If that assumption is incorrect, then it's a different story of course. Let me know.

There's very little effort to just add a new function and leave the old as is, but it will be pure dead code, not sure which one is better.

Reply:
@jagpreetsinghsasan Could you take over the review of this PR, please?

A side note regarding `verifyValidatorJwt` that is a matter of discussion here, I want to extract it into separate package in a commit I'm working on right now, to keep the validator auth logic in one place. Would be useful to decide whether it's OK to treat it as a private interface (which is true from functional perspective). Please check discussion earlier for details :)

Reply:
@outSH Definitely I can have my PR reviews as well. The process of a PR acceptance and merge involves PR review from all the reviewers.
To make the process more manageable, I personally, try to give sequential reviews prior/later to the ongoing reviews, so that the common reviews or clashing reviews don't create further troubles/misunderstanding for the contributor as the involved reviewers.

Reply:
@outSH 

> @petermetz It's not a test-util. It's exported separately to simplify unit test of message verification functionality.

`to simplify unit test of ...` is the very definition of a test utility IMO.

> but testing it through an ApiClient would greatly convolute the test.

I might be missing something. Wouldn't the change be a one liner for updating the import statement within the test case that pulls in the function?

> but it will be pure dead code, not sure which one is better.

We can call the dead code by its pretty name (deprecated) and open an issue to remove it in v2.0.0 (and label the issue with the breaking change label so we know not to work on it prior to 2.0.0 so my vote goes for that.

Reply:
@petermetz 

Separation of component to make it injectable is a _test utility_ for you? OK :)

Could you please clarify what is your definition of a public interface? This function is not exported outside of this package (i.e. not exported in `public-api`, which as the name suggest, should be only public interface definition IMO). It's exported from this file-module just to simplify injection in unit test phase (internally, within a package). Do we want to keep static everything with export word anywhere in the project?